packages/riverpod_analyzer_utils_tests/test/ref_invocation_test.dart

```diff
@@ -411,6 +411,76 @@ final provider = Provider<int>((ref) {
     );
   });
 
+  testSource('Decodes nested ref.read invocations with family providers',
+      runGenerator: true, source: '''
+import 'package:riverpod/riverpod.dart';
+import 'package:riverpod_annotation/riverpod_annotation.dart';
+
+part 'foo.g.dart';
+
+final dep = FutureProvider((ref) => 0);
+final dep2 = FutureProvider.family((ref, int arg) => 0);
+final dep3 = FutureProvider.family((ref, int arg) => 0);
+
+final provider = Provider<int>((ref) {
+  ref.read(dep2(ref.read(dep)));
```

Comment:
Another scenario I'm thinking of is:

```dart
ref.whatever(function(ref.whatever));
```

The indirection could possibly break this logic

Reply:
I've added a test for this now.